library.js

```diff
@@ -35,13 +35,18 @@ exports.load = function(params, callback) {
 
       // error meaning, session not found
       if (error) {
-        //req.uid exists meaning, musicoin session is invalidated and forum session not
-        if (req.uid) {
-          // invalidate forum session too
-          req.logout();
-          // return res.redirect('/');
+        if (err.message === 'INVALID_EMAIL') {
+          res.redirect('/email_not_found');
```

Comment:
Exit here, in case of invalid email error. We do not want the rest of the code to execute in such cases.